types/rox-node/index.d.ts

```diff
@@ -0,0 +1,236 @@
+ *
+ * https://support.rollout.io/docs/nodejs-api#section-setup
+ */
+export function setup(apiKey: string, options?: RoxSetupOptions): Promise<unknown>;
+
+export interface RoxSetupOptions {
+  version?: string;
+  // https://support.rollout.io/docs/configuration-fetched-handler
+  configurationFetchedHandler?(fetcherResult: RoxFetcherResult): void;
+  debugLevel?: 'verbose';
+  // https://support.rollout.io/docs/nodejs-api#section-using-the-impressionhandler-option
+  impressionHandler?(reporting: RoxReporting, experiment: RoxExperiment, context: unknown): void;
+  platform?: string;
+  fetchIntervalInSec?: number;
+  disableNetworkFetch?: boolean;
+  devModeSecret?: string;
+}
+
+export enum RoxFetcherStatus {
+  AppliedFromEmbedded = 'APPLIED_FROM_EMBEDDED',
+  AppliedFromCache = 'APPLIED_FROM_CACHE',
+  AppliedFromNetwork = 'APPLIED_FROM_NETWORK',
+  ErrorFetchFailed = 'ERROR_FETCH_FAILED'
+}
+
+export interface RoxFetcherResult {
+  fetcherStatus: RoxFetcherStatus;
+  creationDate: Date;
+  hasChanges: boolean;
+  errorDetails?: string;
+}
+
+export interface RoxReporting {
+  name: string;
+  value: string;
+}
+
+export interface RoxExperiment {
+  identifier: string; //  experiment id
+  name: string;
+  isArchived: boolean;
+  labels: string[]; // experiment's labels. assigned from dashboard
+}
+
+/**
+ * https://support.rollout.io/docs/custom-properties
+ */
+export function setCustomNumberProperty(name: string, value: number | (() => number)): void;
+export function setCustomStringProperty(name: string, value: string | (() => string)): void;
+export function setCustomBooleanProperty(name: string, value: boolean | (() => boolean)): void;
+export function setDynamicCustomPropertyRule(
+  handler: (propName: string, context: unknown) => number | string | boolean
+): void;
+
+/**
+ * Pulls the latest configuration and flag values down from the Rollout servers
+ *
+ * https://support.rollout.io/docs/nodejs-api#section-fetch
+ */
+export function fetch(): Promise<unknown>;
+
+/**
+ * Creates a new Flag
+ * https://support.rollout.io/docs/nodejs-api#section--flag-
+ */
+export class Flag {
+  constructor(defaultValue?: boolean);
+
+  // The name of the Flag
+  readonly name: string;
+
+  // Default value of the Flag
+  readonly defaultValue: boolean;
+
+  // Returns true when the flag is enabled
+  isEnabled(): boolean;
+
+  getNameDetails(): RoxNameDetails;
+}
+
+/**
+ * Used to create and manage Rollout feature flags that determine different predefined values
+ *
+ * https://support.rollout.io/docs/nodejs-api#section--variant-
+ */
+export class Variant<T extends string = string> {
+  constructor(defaultValue: T, options: ReadonlyArray<T>, name?: string);
+
+  // The name of the Variant
+  readonly name: string;
+
+  // Default value of the Variant
+  readonly defaultValue: BasicType<T>;
+
+  // Returns the current value of the Variant, accounting for value overrides
+  getValue(): BasicType<T>;
+
+  getNameDetails(): RoxNameDetails;
```

Comment:
This should be a private function.

please do not document it
please rename it (add _)